### DISCUSSION TITLE
graceful shutdown don't work

### DISCUSSION THREAD

**Discussion by luhea** [trinodb/trino#18508] [Q&A]
Trino :421,deloy in k8s,turn on LDAP auth and passworld file auth.
curl -v -X PUT -d '"SHUTTING_DOWN"' -H "Content-type: application/json" http://worker:8080/v1/info/state
result:Basic authentication or X-Trino-User must be sent
curl -v -X PUT -d '"SHUTTING_DOWN"' -H "Content-type: application/json" --header "X-Trino-User:test" http://worke r:8080/v1/info/state
result: HTTP/1.1 403 Forbidden
i add http-server.authentication.header.user-mapping.pattern=(.*)(@.*) in config.properties  restart error:
Configuration property 'http-server.authentication.header.user-mapping.pattern' was not used

what should i do ?

**Comment by hashhar** [comment]
If you're using LDAP/password file auth then you need to pass the proper credentials as well via the `Authorization` header.

**Comment by hafizmujadidKhalid** [comment]
@hashhar !
I have same issue, I did pass the right credentials but it still gives me 401. Coordinator has password authentication enabled and i can query using same username/password. But if i try graceful shutdown API on worker, then i get 401.

`curl -v  -X PUT -d "SHUTTING_DOWN" -H "Content-type: application/json" -H "Authorization: Basic XXXXXXXXXXXXX" http://localhost:8080/v1/info/state`

I get following output:

> *   Trying 127.0.0.1:8080...
> * Connected to localhost (127.0.0.1) port 8080 (#0)
> PUT /v1/info/state HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/7.81.0
> Accept: */*
> Content-type: application/json
> Authorization: Basic XXXXXXXXXXXXX
> Content-Length: 15
> 
> * Mark bundle as not supporting multiuse
> HTTP/1.1 401 Unauthorized
> Date: Thu, 14 Dec 2023 23:43:26 GMT
> Content-Type: text/plain;charset=utf-8
> WWW-Authenticate: Basic realm="Trino"
> Content-Length: 48 
> * Connection #0 to host localhost left intact

**Reply by hashhar** [reply]
see also https://github.com/trinodb/trino/issues/20014 if you have access control configured
